parsing/lexer.mll

```diff
@@ -115,12 +115,23 @@ let keyword_table =
   let tbl = Hashtbl.create 149 in
   refresh_keywords tbl;
   tbl
-
 let keyword_edition v =
-  Option.iter (fun v ->
-    if v < (5, 3) then
-      Hashtbl.remove keyword_table "effect"
-  ) v
+  if v < (3,7) then begin
+      Hashtbl.add keyword_table "parser" None;
+      if v < (1,0) then (
+        List.iter (Hashtbl.remove keyword_table)
+          ["class"; "inherit";"method";"new";"private";"constraint";"private";
+           "virtual"]
+      );
+      if v < (1,6) then (
+        Hashtbl.remove keyword_table "lazy";
+        Hashtbl.remove keyword_table "assert"
+      )
+    end;
+  if v < (4,2) then
+    Hashtbl.remove keyword_table "nonrec";
+  if v < (5, 3) then
+    Hashtbl.remove keyword_table "effect"
```

Comment:
Note: if you do refactor this code to simply return a list/set of supported keywords, you will probably rewrite this piece of code, and maybe the result will be nicer. (Ideally the natural approach reads like keywords from newer versions are added, rather than removed.)